packages/data-stores/src/site/reducer.ts

```diff
@@ -0,0 +1,79 @@
+/**
+ * External dependencies
+ */
+import { Reducer } from 'redux';
+import { combineReducers } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { ActionType, NewSite, NewSiteErrorResponse } from './types';
+import * as Actions from './actions';
+
+const newSiteData: Reducer<
+	NewSite | {} | undefined,
+	| ReturnType< typeof Actions[ 'receiveNewSite' ] >
+	| ReturnType< typeof Actions[ 'receiveNewSiteFailed' ] >
+> = ( state = undefined, action ) => {
+	if ( action.type === ActionType.RECEIVE_NEW_SITE ) {
+		const { response } = action;
+		return {
+			...response.blog_details,
+		};
+	} else if ( action.type === ActionType.RECEIVE_NEW_SITE_FAILED ) {
+		return undefined;
+	}
+	return state;
+};
+
+const newSiteError: Reducer<
+	NewSiteErrorResponse | undefined,
+	| ReturnType< typeof Actions[ 'fetchNewSite' ] >
+	| ReturnType< typeof Actions[ 'receiveNewSite' ] >
+	| ReturnType< typeof Actions[ 'receiveNewSiteFailed' ] >
+> = ( state = undefined, action ) => {
+	switch ( action.type ) {
+		case ActionType.FETCH_NEW_SITE:
+			return undefined;
+		case ActionType.RECEIVE_NEW_SITE:
+			return undefined;
+		case ActionType.RECEIVE_NEW_SITE_FAILED:
+			return {
+				error: action.error.error,
+				status: action.error.status,
+				statusCode: action.error.statusCode,
+				name: action.error.name,
+				message: action.error.message,
+			};
+	}
+	return state;
+};
+
+const isFetchingSite: Reducer<
```

Comment:
I'd discourage this manual request tracking. We should be able to find core-provided functionality for this.

Reply:
The way I understand it so far (very limited), is that `isResolving` is a method limited to resolvers (as side-effect for [selectors](https://github.com/WordPress/gutenberg/blob/f63053cace3c02e284f00918e1854284c85b9132/packages/data/src/store/index.js#L16)). 

At the moment we've designed `createSite` as an action, and not a selector. 

Intuitively this seems correct as we are making changes and not retrieving data.

Do you think there's an elegant way to go about this other than via the reducer?

Middleware using [redux-routine](https://developer.wordpress.org/block-editor/packages/packages-redux-routine/) maybe?

Reply:
@andrewserong I think we could park the `isFetching` debate until we work out the best way to do it. 

We'll definitely need some sort of `isRequesting` flag, particularly in light of https://github.com/Automattic/wp-calypso/issues/39268

For now, we could probably open this PR up for proper review and get it merged.

After that we can decide how to integrate it into Frankenflow.

👍